src/mds/SimpleLock.h

```diff
@@ -210,17 +209,33 @@ class SimpleLock {
   void encode_locked_state(ceph::buffer::list& bl) {
     parent->encode_lock_state(type->type, bl);
   }
+
+  using waitmask_t = MDSCacheObject::waitmask_t;
+  static constexpr auto WAIT_ORDERED = waitmask_t(MDSCacheObject::WAIT_ORDERED);
+  int get_wait_shift() const;
+  MDSCacheObject::waitmask_t getmask(uint64_t mask) const {
```

Comment:
This looks great! My only wish is that we make the calculations and assert once in the constructor, and have the 128bit prepared clean mask base ready to be ORed with the actual wait bit before passing it down to the waiting list, as the shift won't ever change for a given lock. Otherwise, we're running this every time a waiter is added, tested, or taken. For your consideration.

Reply:
That is more efficient computationally but we would need to store the shift in the `SimpleLock` which amplifies the space use of `SimpleLock`. We should try to avoid that.

(Alternatively, we could sub-class `SimpleLock` and its derivatives based on the actual lock type (`FileLock`, etc.). Then we could use a method to return the shift.

Reply:
Ack